testing/test_pytester.py

```diff
@@ -847,3 +847,14 @@ def test_testdir_makefile_ext_empty_string_makes_file(testdir) -> None:
     """For backwards compat #8192"""
     p1 = testdir.makefile("", "")
     assert "test_testdir_makefile" in str(p1)
+
+
+def test_pytester_assert_outcomes_warnings(pytester: Pytester) -> None:
+    p = pytester.makepyfile(
+        """
+        def test_with_warning():
+            pass
+        """
+    )
+    result = pytester.runpytest(p, "--strict")
```

Comment:
Hmm let's improve this by making the test itself generate the warning:

```pyhton
import warnings

def test_with_warning():
    warnings.warn(UserWarning("some custom warning"))
```

Let's not rely on `--strict` being deprecated, because then this test will break once we remove `--strict` and will need to adapt it anyway.

Reply:
thanks @nicoddemus see my other comment, you read my mind :)

Reply:
Ahh we posted the comments at the same time, it seems. 😁

Reply:
hmm, this approach does not write `1 warnings`, am i missing a flag to display that?

Reply:
when i use your approach with pytester, the test fails w/o warning, is there a way to get the outcome from running a test w/o running `pytester.runpytest()` ?

Reply:
Ahh right, what happens is that we change all warnings into errors when running the pytest suite.

You can override that applying `@pytest.mark.filterwarnings("default")` to `test_pytester_assert_outcomes_warnings`:

```python
@pytest.mark.filterwarnings("default")
def test_pytester_assert_outcomes_warnings(pytester: Pytester) -> None:
    p = pytester.makepyfile(
        """
        import warnings
        def test_with_warning():
            warnings.warn(UserWarning("some warning"))
        """
    )
    result = pytester.runpytest(p)
    result.assert_outcomes(passed=1, warnings=1)
```

Reply:
every day is a school day, thanks